README.md

```diff
@@ -0,0 +1,163 @@
+# near-lake
+
+**NB! The project is in early stage of development and shouldn't be used in production environmental yet**
+
+NEAR Lake is an indexer built on top of [NEAR Indexer microframework](https://github.com/nearprotocol/nearcore/tree/master/chain/indexer)
+to watch the network and store all the events as JSON files on AWS S3.
+
+## Concept
+
+We used to have [NEAR Indexer for Explorer](https://github.com/near/near-indexer-for-explorer) that was watching for 
+the network and stored all the events to PostgreSQL database. PostgreSQL became the main bottleneck for us. After some
+brainstorming sessions and researches we decided to go with [SingleStore](https://www.singlestore.com/) database. 
+
+Knowing the fact that [NEAR Explorer](https://explorer.near.org) is not the only project that uses the Indexer for Explorer's 
+database, we wanted to come up with the concept that will allow us to cover even more projects that can benefit from the data 
+from NEAR Protocol.
+
+That's why we decided to store the data from the blockchain as JSON files on AWS S3 bucket that can be used
+as a data source for different projects. 
+
+As "Indexer for Explorer Remake" project we are going to have `near-lake` as a data writer. There's going to be
+another project that will read from AWS S3 bucket and will store all the data in SingleStore database. This
+will replace NEAR Indexer for Explorer PostgreSQL database at some moment and will become the main
+source for NEAR Explorer.
+
+## How to start
+
+The final setup consists of the following components:
+* AWS S3 Bucket as a storage
+* NEAR Lake binary that operates as a regular NEAR Protocol peer-to-peer node, so you will operate it as 
+  any other [Regular/RPC Node in NEAR](https://docs.near.org/docs/develop/node/rpc/hardware-rpc)
+
+### Prepare Development Environment
+
+Before you proceed, make sure you have the following software installed:
+* [Rust compiler](https://rustup.rs/) of the version that is mentioned in `rust-toolchain` file in the root of
+  [nearcore](https://github.com/nearprotocol/nearcore) project.
+* Ensure you have [AWS Credentials configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
+    From AWS Docs:
+
+  > For example, the files generated by the AWS CLI for a default profile configured with aws configure looks similar to the following.
+  >
+  > ~/.aws/credentials
+  > ```
+  > [default]
+  > aws_access_key_id=AKIAIOSFODNN7EXAMPLE
+  > aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+  > ```  
+  >  
+  > ~/.aws/config
+  > ```
+  > [default]
+  > region=us-west-2
+  > output=json
+  > ```  
+
+### Compile NEAR Lake
+
+```bash
+$ cargo build --release
+```
+
+### Configure NEAR Lake
+
+To connect NEAR Lake to the specific chain you need to have necessary configs, you can generate it as follows:
+
+```bash
+$ ./target/release/near-lake --home-dir ~/.near/testnet init --chain-id testnet --download-config --download-genesis
```

Comment:
We previously used `--home-dir` in our indexers, but `neard` uses just `--home`. I suggest we go with `--home` in `near-lake`, so we can mostly reflect the command line of `neard` and avoid unnecessary conversations going forward.